### DISCUSSION TITLE
How to use the LoRaWAN_TTN.ino file?

### DISCUSSION THREAD

**Discussion by isaacmateosv** [ropg/heltec_esp32_lora_v3#29] [General]
Hi there! Just wondering how I enter the parameters required for connecting my Heltec Wireless Stick V3 to a LoRa Gateway. 

When I try to put the region, I got "Error: 'US915' is not a supported band."

Am I missing stuff that I need to review? Please, help.

Thanks in advance.

**Comment by ropg** [comment]
That's odd, US915 should be recognized just fine. Are you sure there's no hidden control characters, backspaces, funny unicode characters or something in there? The routine to read the serial data is kind of rudimentary, but other than that the code to recognize the band names is really straightforward. What happens when you enter another band, like EU868 ?

**Reply by isaacmateosv** [reply]
I didn't see your reply! Thanks for replying.

I have tried many things but it's still not ok. I can't connect my board to a Multitech Conduit AP gateway.

I will add a screenshot of the Arduino IDE settings if you would like to help me. Sorry for taking so long to answer and thanks in advance, again.
![image](https://github.com/ropg/heltec_esp32_lora_v3/assets/19366181/366f5967-0f57-4b01-9d34-bf9720f9765a)

**Reply by isaacmateosv** [reply]
BTW, this is the verbose in the Arduino serial monitor:
[verbose.txt](https://github.com/user-attachments/files/16033355/verbose.txt)

**Reply by isaacmateosv** [reply]
Turns out there was indeed a line break. Thanks again for your time. 😅

**Reply by ropg** [reply]
Alright. Mystery solved. Hmm, maybe I should strip that string.

**Reply by ropg** [reply]
If you use the version of [**LoRaWAN_ESP32**](https://github.com/ropg/LoRaWAN_ESP32) that's now on Github you'll see that any [control chars are now ignored](https://github.com/ropg/LoRaWAN_ESP32/commit/b6072b11239fa12cbab091ad0985c23fce10fc9e). Will find its way into next release version soon.

**Reply by isaacmateosv** [reply]
Thanks again for all your support in this topic. I have another question though, if you wish to provide further help.
What should I enter in the "nwkKey" parameter if it is the first time I am linking the node to a gateway? (I mean, I have no session recorded in the gateway). I was using JoinEUI as 00... 00 and I have got the DevEUI from the manufacturer (nothing else).
I tried generating and using my own keys, but the gateway says it is an unknown device and no package can be sent successfully. 

I understand that OTAA connection does not require the nwkKey parameter. What am I doing wrong?

The LoRa gateway is a Multitech Conduit AP.

Thanks in advance again, Mr. ropg.

**Reply by ropg** [reply]
I really only have enough experience playing with The Things Network. There you create an endpoint and it will let you set (or generate for you) the provisioning data, which includes nwkKey. Your gateway is essentially transparent, you don't need its devEUI, you just need to set the same ones on your endpoint as at TTN. (Or maybe I don't quite understand your issue.)
<img width="749" alt="image" src="https://github.com/ropg/heltec_esp32_lora_v3/assets/343196/dd83bbb1-8eb6-46f9-a620-543f82bf703a">
